### PR TITLE
Move S.R.Intrinsics.X86 to shared framework

### PR DESCRIPTION
**Pull request by fiigii**
See the discussion in #24346 for details.

### CODE REVIEW

**Comment by fiigii** [comment]
@weshaggard @joperezr @jkotas PTAL

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 1]
```diff
@@ -3579,7 +3579,9 @@
       }
     },
     "System.Runtime.Intrinsics.X86": {
-      "InboxOn": {},
+      "InboxOn": {
+        "netcoreapp2.1": "4.0.0.0"
+      },
```
Comment (fiigii) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: The build system requires  "InboxOn". Is this version right?

Comment (weshaggard) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Yes it should be.

**Comment by weshaggard** [comment]
Thanks @fiigii.

We are currently having issues with CI that are not related to your changes. Hopefully those will be fixed soon.

**Comment by fiigii** [comment]
@dotnet-bot test this please

**Comment by fiigii** [comment]
All tests passed, could you please merge this? @weshaggard @joperezr

**Comment by weshaggard** [comment]
Thanks @fiigii.

**Pull request merged**